daemon/containerd/image.go

```diff
@@ -122,6 +122,36 @@ func (i *ImageService) GetImage(ctx context.Context, refOrID string, options ima
 	return img, nil
 }
 
+func (i *ImageService) GetImageVariant(ctx context.Context, refOrID string, options imagetype.GetImageOpts) (*ocispec.Descriptor, error) {
+	cs := i.client.ContentStore()
+
+	desc, err := i.resolveDescriptor(ctx, refOrID)
+	if err != nil {
+		return nil, nil
+	}
+
+	platform := platforms.AllPlatformsWithPreference(cplatforms.Default())
+	if options.Platform != nil {
+		platform = cplatforms.OnlyStrict(*options.Platform)
```

Comment:
This one should be `cplatforms.Only` to match `PrepareSnapshot` behavior. We want the "lower" variant images to be usable on "higher" variant platforms (example: `linux/arm/v5`  will run fine on `linux/arm/v6`).

Reply:
Hah, high five - same thoughts at the same time (https://github.com/moby/moby/pull/44958#discussion_r1123631350). :sunglasses:

Reply:
Oh, indeed! I didn't notice it at the time I was posting my comment 😆

Reply:
Ooh, missed that, thanks ❤️